plugins/destination/postgresql/client/spec/spec.go

```diff
@@ -0,0 +1,67 @@
+package spec
+
+import (
+	_ "embed"
+	"errors"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/v4/configtype"
+	"github.com/invopop/jsonschema"
+)
+
+const (
+	defaultBatchSize      = 10000
+	defaultBatchSizeBytes = 100000000
+	defaultBatchTimeout   = 60 * time.Second
+)
+
+type Spec struct {
+	// Connection string to connect to the database. This can be a URL or a DSN, as per [`pgxpool`](https://pkg.go.dev/github.com/jackc/pgx/v4/pgxpool#ParseConfig)
+	//
+	// - `"postgres://jack:secret@localhost:5432/mydb?sslmode=prefer"` _connect with tcp and prefer TLS_
+	// - `"postgres://jack:secret@localhost:5432/mydb?sslmode=disable&application_name=pgxtest&search_path=myschema&connect_timeout=5"` _be explicit with all options_
+	// - `"postgres://localhost:5432/mydb?sslmode=disable"` _connect with os username cloudquery is being run as_
+	// - `"postgres:///mydb?host=/tmp"` _connect over unix socket_
+	// - `"dbname=mydb"` _unix domain socket, just specifying the db name - useful if you want to use peer authentication_
+	// - `"user=jack password=jack\\'ssooper\\\\secret host=localhost port=5432 dbname=mydb sslmode=disable"` _DSN with escaped backslash and single quote_
+	ConnectionString string `json:"connection_string,omitempty" jsonschema:"required,minLength=1"`
+
+	// Available: `error`, `warn`, `info`, `debug`, `trace`.
+	// Defines what [`pgx`](https://github.com/jackc/pgx) call events should be logged.
+	PgxLogLevel LogLevel `json:"pgx_log_level,omitempty" jsonschema:"default=error"`
+
+	// Maximum number of items that may be grouped together to be written in a single write.
+	BatchSize int `json:"batch_size,omitempty" jsonschema:"minimum=1,default=10000"`
+
+	// Maximum size of items that may be grouped together to be written in a single write.
+	BatchSizeBytes int `json:"batch_size_bytes,omitempty" jsonschema:"minimum=1,default=100000000"`
+
+	// Maximum interval between batch writes.
+	BatchTimeout configtype.Duration `json:"batch_timeout,omitempty" jsonschema:"default=60s"`
```

Comment:
Specifying `default` here is ineffective, modify it in JSONSchemaExtend

Reply:
Done

Reply:
9a9b6b0b433d2082e89d880c174eb556b43d99ad